addons/mrp_subcontracting_dropshipping/models/stock_warehouse.py

```diff
@@ -15,6 +15,54 @@ class StockWarehouse(models.Model):
         'stock.rule', 'Subcontracting-Dropshipping MTS Rule'
     )
 
+    @api.model_create_multi
+    def create(self, vals_list):
+        res = super().create(vals_list)
+        # if new warehouse has resupply enabled, enable global route
+        if any(['subcontracting_dropshipping_to_resupply' in vals for vals in vals_list]):
+            res.update_global_route_dropship_subcontractor()
+        return res
+
+    def write(self, vals):
+        res = super().write(vals)
+        # if all warehouses have resupply disabled, disable global route, until its enabled on a warehouse
+        if 'subcontracting_dropshipping_to_resupply' in vals or 'active' in vals:
+            self._update_dropship_subcontract_rules()
```

Comment:
This needs the same logic for when to update the rules as the write in mrp_subcontracting module for stock.warehouse (i.e. what I wrote in my other comment).